### PR TITLE
MultiDimensional Media Mix Model

### PR DESCRIPTION
**Pull request by cetagostini**
## Description
Creating an API to support multiple dims.

## Related Issue

- [ ] Closes #
- [x] Related to #497 #460 

## Checklist

- [ ] Checked that [the pre-commit linting/style checks pass](https://docs.pymc.io/en/latest/contributing/python_style.html)
- [ ] Included tests that prove the fix is effective or that the new feature works
- [ ] Added necessary documentation (docstrings and/or example notebooks)
- [ ] If you are a pro: each commit corresponds to a [relevant logical change](https://wiki.openstack.org/wiki/GitCommitMessages#Structural_split_of_changes)

## Modules affected

- [x] MMM
- [ ] CLV

## Type of change

- [x] New feature / enhancement
- [ ] Bug fix
- [ ] Documentation
- [ ] Maintenance
- [ ] Other (please specify):

----
📚 Documentation preview 📚: https://pymc-marketing--1004.org.readthedocs.build/en/1004/

### CODE REVIEW

File: pymc_marketing/mmm/MultiDimensionalMMM.py
Comment (wd60622) [inline pymc_marketing/mmm/MultiDimensionalMMM.py:1]: Change file names for python module convention

File: pymc_marketing/mmm/MultiDimensionalMMM.py [hunk 1]
```diff
@@ -0,0 +1,591 @@
...
+from pymc_marketing.model_config import parse_model_config
+from pymc_marketing.prior import Prior
+
+
+class VanillaMultiDimensionalMMM:
+    """Docstring example."""
+
+    _model_name: str = "BaseMMM"
+    _model_type: str = "BaseValidateMMM"
+    version: str = "0.0.3"
```
Comment (wd60622) [inline pymc_marketing/mmm/MultiDimensionalMMM.py:45]: Why this version?

File: pymc_marketing/mmm/MultiDimensionalMMM.py
Comment (wd60622) [inline pymc_marketing/mmm/MultiDimensionalMMM.py:1]: Seems like a lot of duplicated code

**Review by wd60622 — COMMENTED** [review]
Initial look

File: pymc_marketing/mmm/MultiDimensionalMMM.py
Comment (wd60622) [reply pymc_marketing/mmm/MultiDimensionalMMM.py:1]: Seeing that much of the code is the same (with these multi-dimensional tweaks), I think this is a big reason to have them in a single class.

I think my major concern with this is that plotting, (possible) budget optimizer, and some other methods will likely not supported out of the box.

File: pymc_marketing/mmm/MultiDimensionalMMM.py (hunk 1 above)
Comment (cetagostini) [reply pymc_marketing/mmm/MultiDimensionalMMM.py:45]: Just a copy-paste of the original code!

File: pymc_marketing/mmm/MultiDimensionalMMM.py
Comment (cetagostini) [reply pymc_marketing/mmm/MultiDimensionalMMM.py:1]: > I think my major concern with this is that plotting, (possible) budget optimizer, and some other methods will likely not supported out of the box.

Correct, I raise the issue about other methods being not compatible. I don't have any strong opinion here, but I was thinking on create a separate class for plots which can handle the same multi-dimensional tweaks (Same for budget optimizer).

Maybe @OriolAbril has ideas about this?

You can take look to my ideas [here](https://discord.com/channels/745261709622771773/948587000464834580/1281260228498817195). Here I was proposing a way manage the transition between the meantime.

Comment (cetagostini) [reply pymc_marketing/mmm/MultiDimensionalMMM.py:1]: > Seeing that much of the code is the same (with these multi-dimensional tweaks), I think this is a big reason to have them in a single class.

Totally, like that but we need to prepare a transition period ✌🏻

**Comment by juanitorduz** [comment]
@cetagostini shall we start another clean Pr for this? The idea looks very promising 🙏

**Pull request closed without merging**